### PR TITLE
Add module metal_ssh_key, docs and tests

### LINKED ISSUE #68: metal_ssh_key
Create module for metal_ssh_key.

Analogical to resource equinix_metal_ssh_key in terraform-provider-equinix:

- https://registry.terraform.io/providers/equinix/equinix/latest/docs/resources/equinix_metal_ssh_key
- https://github.com/equinix/terraform-provider-equinix/tree/master/equinix/resource_metal_ssh_key.go

Module should
- be in `plugins/modules/metal_ssh_key.py`
- follow [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
- be [idempotent](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency>)
- support `check_mode` if it's an `_info` module
- have doc fields for (ansible-specdoc)[https://github.com/linode/ansible-specdoc] (see [metal_project.py](https://github.com/equinix-labs/ansible-collection-equinix/blob/main/plugins/modules/metal_project.py) for an example
  - module_spec
  - specdoc_example
  - result_sample
  - SPECDOC_METAL
  - (DOCUMENATION, EXAMPLES and RETURN will be generated by `make docs`)
- have an integration test in `tests/integration/targets/metal_ssh_key/tasks/main.yml`
- (semi-optional) pass sanity test: 
  `$ cd .ansible/collections/ansible_collections/equinix/cloud`
  `$ ansible-test sanity -v --docker --python 3.10 metal_ssh_key.py`

See the newest added module to get an idea how to do implement this.

### PR DESCRIPTION
**Pull request by t0mk**
fixes #68

### CODE REVIEW

**Comment by t0mk** [comment]
fixes #68

File: docs/modules/metal_ssh_key.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+```
+
+```yaml
+- name: Create new ssh_key within non - default organization
+  hosts: localhost
+  tasks:
+  - equinix.cloud.metal_ssh_key:
+      name: "my org ssh_key"
+      organization_id: "a4cc87f9-e00f-48c2-9460-74aa60beb6b0"
```
Comment (ctreatma) [inline docs/modules/metal_ssh_key.md:27]: Is this a valid configuration, or should `organization_id` be `project_id`?  I thought there were only 2 types of SSH keys:
- user SSH keys, which have access to every project the user belongs to, regardless of org
- project SSH keys, which belong to a single project and only have access to that project

Comment (t0mk) [reply docs/modules/metal_ssh_key.md:27]: Hmm, yes, that's not right. Also I see that there are `instance_ids` to inject the keys to existing instances. Will rework.

https://github.com/equinix-labs/metal-python/blob/72eaf28f75e2107aa75e6bd71d333ab4f030bdf9/equinix_metal/equinix_metal/models/ssh_key_create_input.py#L26

File: docs/modules/metal_ssh_key.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# metal_ssh_key
+
+Manage ssh_keys in Equinix Metal. You can use *id* or *label* to lookup a ssh_key. If you want to create new ssh_key, you must provide *name*.
```
Comment (ctreatma) [inline docs/modules/metal_ssh_key.md:3]: I think it'd be good here to call out that this module creates personal SSH keys, and to link to docs that cover the difference: https://deploy.equinix.com/developers/docs/metal/accounts/ssh-keys/#personal-keys-vs-project-keys

File: docs/modules/metal_ssh_key_info.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# metal_ssh_key_info
+
+Gather user SSH key resources
```
[the remark this answers is not in the document]
Comment (t0mk) [reply docs/modules/metal_ssh_key_info.md]: fixed it in 0f149cd

File: docs/modules/metal_ssh_key.md (hunk 2 above)
Comment (t0mk) [reply docs/modules/metal_ssh_key.md]: fixed it in https://github.com/equinix-labs/ansible-collection-equinix/commit/0f149cdef66bd323370a6cdec67724770c8e813c

**Review by cprivitere — APPROVED** [review]

**Pull request merged**